service/model/project.go

```diff
@@ -138,3 +155,45 @@ func (c *ReleaseNotificationConfig) Update(u UpdateReleaseNotificationConfigInpu
 		c.ShowSourceCode = *u.ShowSourceCode
 	}
 }
+
+// toGithubRepository parses a GitHub repository URL and returns a GithubRepository struct
+// Github Repository URL must be in the format: http(s)://github.com/owner/repo or http(s)://www.github.com/owner/repo
+func toGithubRepository(rawURL string) (GithubRepository, error) {
+	if rawURL == "" {
+		return GithubRepository{}, nil
+	}
+
+	u, err := url.Parse(rawURL)
+	if err != nil {
+		return GithubRepository{}, errProjectInvalidGithubRepositoryURL
+	}
+
+	if !validator.IsAbsoluteURL(rawURL) {
+		return GithubRepository{}, errProjectInvalidGithubRepositoryURL
+	}
+
+	if u.Host != githubHost && u.Host != githubHostWWW {
```

Comment:
Isn't this a bit of an excessive check? If they want to break it, they can do that anyway by providing a URL to a non-existing repository, just starting with `github.com`.

Reply:
Thank you, that's true. If the check was more complicated, I would definitely remove it, as it isn't worth it since it can be easily broken.

But it's a very simple check, it can still effectively handle basic cases when someone might accidentally provide the wrong URL (not intentionally). And it is already there so I keep it for now.